db_sql.go

```diff
@@ -469,89 +539,155 @@ func SqlCreateDatabase(conx *DBConnection) error {
+		ALTER TABLE authsession ADD COLUMN userid BIGSERIAL;
+		CREATE INDEX idx_authsession_userid ON authsession (userid);
+		INSERT INTO authsession (id, sessionkey, userid, permit, expires)
+		SELECT authtemp.id, authtemp.sessionkey, us.userid, authtemp.permit, authtemp.expires FROM authsession_temp AS authtemp
+		JOIN userstore AS us ON us.identity = authtemp.identity;
+		DROP TABLE IF EXISTS authsession_temp;
+
+		CREATE TABLE authuser_temp(id BIGSERIAL PRIMARY KEY, identity VARCHAR, password VARCHAR, permit VARCHAR, pwdtoken VARCHAR);
+		INSERT INTO authuser_temp (id, identity, password, permit, pwdtoken) SELECT * FROM authuser;
+		TRUNCATE authuser;
+		DROP INDEX idx_authuser_identity;
+		ALTER TABLE authuser DROP COLUMN identity;
+		ALTER TABLE authuser ADD COLUMN userid BIGSERIAL;
+		ALTER TABLE authuser ADD CONSTRAINT fk_storeUserId_authUserId FOREIGN KEY (userid) REFERENCES userstore (userid);
+		INSERT INTO authuser (id, userid, password, permit, pwdtoken)
+		SELECT usertemp.id, us.userid, usertemp.password, usertemp.permit, usertemp.pwdtoken FROM authuser_temp AS usertemp
+		JOIN userstore AS us ON us.identity = usertemp.identity;
+		DROP TABLE IF EXISTS authuser_temp;`,
+	}
+
+	for _, src := range text {
+		srcCapture := src
+		migrations = append(migrations, func(tx migration.LimitedTx) error {
+			_, err := tx.Exec(srcCapture)
+			return err
+		})
+	}
+	return migrations
+}
+
+//This function makes sure that the old migrations that were part of the old migration tool have run.
+func runBootstrap(db *sql.DB) (bool, error) {
+	oldVersionTables := []string{"authgroup_version", "authsession_version", "authuser_version"}
+	// We first want to see if the new migrator has taken over, by checking if old version tracking tables exist
+	tableNameResult := ""
+	query := db.QueryRow(fmt.Sprintf("SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = '%v'", oldVersionTables[0]))
+	err := query.Scan(&tableNameResult)
+	if err != nil {
+		if strings.Index(err.Error(), "no rows in result set") != -1 {
+			return true, nil
 		}
+		return false, err
+	}
+	// Tracking tables dont exist, this is up to date
+	if tableNameResult == "" {
+		return true, nil
 	}
-}
 
-// Create a Postgres DB schema necessary for a Session database
-func SqlCreateSchema_Session(conx *DBConnection) error {
-	versions := make([]string, 0)
-	versions = append(versions, `
-	CREATE TABLE authsession (id BIGSERIAL PRIMARY KEY, sessionkey VARCHAR, identity VARCHAR, permit VARCHAR, expires TIMESTAMP);
-	CREATE UNIQUE INDEX idx_authsession_token    ON authsession (sessionkey);
-	CREATE        INDEX idx_authsession_identity ON authsession (identity);
-	CREATE        INDEX idx_authsession_expires  ON authsession (expires);`)
-	// we made some changes that are not backwards compatible, need to force a logout for all users
-	versions = append(versions, `DELETE FROM authsession;`)
+	//
+	checkTableVersion := func(table string, expectedVersion int) bool {
```

Comment:
`check` is usually a bad prefix for a function that returns a bool. You never know whether the return value means `OK` or `Not OK`. In a case like this, I would rather just make a function such as "getTableVersion" that returns (int, error). 
It's also not clear here that a DB-level error is being treated correctly. For example, if the DB connection dies on one of these queries, then checkVersion will return false, which will end up causing the system to try and upgrade the DB schema. This is not right - if one receives an error from the DB, then everything must abort.